src/matching_market.sol

```diff
@@ -207,23 +220,24 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
         return true;
     }
 
-    //set the minimum sell amount for a token
+    //set the minimum sell amount for a token. Uses Uniswap as a price oracle.
     //    Function is used to avoid "dust offers" that have
     //    very small amount of tokens to sell, and it would
     //    cost more gas to accept the offer, than the value
     //    of tokens received.
     function setMinSell(
-        ERC20 pay_gem,     //token to assign minimum sell amount to
-        uint dust          //maker (ask) minimum sell amount
+        ERC20 pay_gem     //token to assign minimum sell amount to
     )
         public
-        auth
         note
         returns (bool)
     {
-        _dust[address(pay_gem)] = dust;
-        emit LogMinSell(address(pay_gem), dust);
-        return true;
+        require(msg.sender == tx.origin, "No indirect calls please");
+        require(address(pay_gem) != DAI, "Can't set dust for DAI");
+        
+        uint256 dust = uniswapSimplePriceOracle.getPriceFor(DAI, address(pay_gem), daiDustLimit);
```

Comment:
I'd suggest that the code of the getPriceFor is directly integrated here. I do not see a very good reason to deploy an intermediate contract and have to add an extra external call. `maker-otc` could implement these functions directly.

Reply:
I see the tests will become a more complex to set, but probably makes sense to save that gas an extra complexity.

Reply:
We discussed it with @maciejka in the past and initially, it was implemented like this. The problem is that it's (almost?) impossible to write tests for it (uniswap library relies on CREATE2 created contracts).

Also it would couple this contract tightly with uniswap which is ugly I think.

Reply:
TBH: I don't consider the gas cost to be an issue here as this will be fired a few times after contract deployment and probably never again...

Reply:
I wanted to write "will become a more complex to **test**"
So yes, I see it will be more complex, I think it can be achieved making a dummy factory which receives a price and manipulates the reserves in order `MatchingMarket` can obtain that one. But yeah it would be more complex.
I agree on coupling part as well, but maybe then we should just call it `OracleLike` from the MatchingMarket perspective and then make the actual implementation a single file with the only functions that are necessary for simpleness.

Reply:
> I think it can be achieved making a dummy factory which receives a price and manipulates the reserves in order MatchingMarket can obtain that one.

I thought so too but to derive pair address you don't call some method of the factory but you rather derive pair address from factories address AND bytecode of the pair contract (CREATE2 works that way). It's implemented by UniswapLibrary. Implemented [here](https://github.com/daifoundation/maker-otc/pull/192/files#diff-f547e6627158bdfe5c3266f90f6dcc74542f7875ae8bd0a49e83cd6a9648c4baR67).

This makes it impossible to test it without actually bringing the real whole Uniswap implementation to this project which is ridiculous IMO. Really, let me know if you know how this can be done the other way. But I already tried different approaches. For now, I will continue will extracting interface.

Reply:
What is more UniswapLibrary and Uniswap requires different solc versions which makes it even more complicated.